src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorProvider.cs

```diff
@@ -52,29 +53,53 @@ public IEnumerable<TagHelperDescriptor> GetTagHelpers(string tagName)
                 return Enumerable.Empty<TagHelperDescriptor>();
             }
 
+            HashSet<TagHelperDescriptor> catchAllDescriptors;
+
             // Ensure there's a HashSet to use.
-            if (!_registrations.TryGetValue(CatchAllDescriptorTarget, out descriptors))
+            if (!_registrations.TryGetValue(CatchAllDescriptorTarget, out catchAllDescriptors))
             {
                 descriptors = new HashSet<TagHelperDescriptor>(TagHelperDescriptorComparer.Default);
             }
-
-            // If the requested tag name is the catch-all target, we should short circuit.
-            if (tagName.Equals(CatchAllDescriptorTarget, StringComparison.OrdinalIgnoreCase))
+            else
             {
-                return descriptors;
+                descriptors = catchAllDescriptors;
             }
 
-            // If we have a tag name associated with the requested name, return the descriptors +
-            // all of the catch-all descriptors.
-            HashSet<TagHelperDescriptor> matchingDescriptors;
-            if (_registrations.TryGetValue(tagName, out matchingDescriptors))
+            // If the requested tag name is the catch-all target, we shouldn't do the work of concatenating extra
+            // descriptors.
```

Comment:
:-1: Not removing. It's been there all along, if we're going to remove it that's a separate issue and we'll do it in another PR, not worth the time imo; as of right now it does no harm.